frontend/src/components/Modal.tsx

```diff
@@ -0,0 +1,113 @@
+import { Dialog, Transition } from '@headlessui/react'
```

Comment:
Added a comment for explanation. The modal needs a focusable element, otherwise it throws an exception. The context is added so that the `ref` can be used both in the `ModalRoot` (to set focus), and `Modal.Body` (to set ref). Otherwise, we would need to manage the `ref` every time a new component that uses `Modal` is created. The context is not needed, but makes the API less error prone